pkg/services/sqlstore/preferences.go

```diff
@@ -120,3 +183,33 @@ func (ss *SQLStore) SavePreferences(ctx context.Context, cmd *models.SavePrefere
 		return err
 	})
 }
+
+func (ss *SQLStore) SavePreferencesJsonData(ctx context.Context, cmd *models.SavePreferencesJsonDataCommand) error {
```

Comment:
Would suggest reuse the `SavePreferences` method for this. Any reason not to? It would however require reading the preferences in the API before saving in case the `JsonData` is not provided and the other way around only `JsonData` provided, but not the other preferences.